tmt/steps/__init__.py

```diff
@@ -324,44 +394,58 @@ def __init__(self, step, data, workdir=None):
             name=data['name'],
             workdir=workdir,
             order=order)
-        self.data = data
+        # It is not possible to use TypedDict here because
+        # all keys are not known at the time of the class definition
+        self.data: Dict[Any, Any] = data
         self.step = step
 
     @classmethod
-    def options(cls, how=None):
+    def base_command(cls, method_class: Optional[Method] = None,
+                     usage: Optional[str] = None) -> click.Command:
+        """ Create base click command (common for all discover plugins) """
+        raise NotImplementedError
+
+    @classmethod
+    def options(cls, how: Optional[str] = None) -> List[click.Option]:
         """ Prepare command line options for given method """
         # Include common options supported across all plugins
+        assert isinstance(tmt.options.verbose_debug_quiet, list)
+        assert isinstance(tmt.options.force_dry, list)
```

Comment:
Please, add a TODO item - these should not be needed once mypy is allowed to peek into other modules.

Reply:
Thanks, added